### PR TITLE
DATAUP-639 first set of updates to dynamic dropdown

### PR DESCRIPTION
**Pull request by briehl**
# Description of PR purpose/changes

This is mostly a cleanup PR for the dynamic dropdown before the major work happens.
It does the following:
* Fixes the dropdown so the default "Loading..." text appears again
* Removes unused dependencies
* Various linting and reorganizing chores.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-639
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook
- [x] Any dependent changes have been merged and published in downstream modules

# Updating Version and Release Notes (if applicable)

- [n/a] [Version has been bumped](https://semver.org/) for each release
- [n/a] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: kbase-extension/scss/partials/_appInput.scss [hunk 1]
```diff
@@ -38,4 +38,21 @@
             }
         }
     }
+
```
Comment (briehl) [inline kbase-extension/scss/partials/_appInput.scss:41]: Moved some inline css (😱 ) here

File: kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js [hunk 2]
```diff
@@ -43,27 +37,24 @@ define([
...
+        baseCssClass = 'kb-appInput__dynDropdown';
 
     function factory(config) {
         const spec = config.parameterSpec,
             runtime = Runtime.make(),
             bus = runtime.bus().connect(),
             channel = bus.channel(config.channelName),
-            dd_options = spec.original.dynamic_dropdown_options || {},
-            dataSource = dd_options.data_source || 'ftp_staging',
+            ddOptions = spec.original.dynamic_dropdown_options || {},
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js:48]: Changed snake_case -> camelCase throughout. I think I got it all.

File: kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js [hunk 3]
```diff
@@ -332,53 +317,77 @@ define([
...
+         * path: "data/bulk/wjriehl/subfolder/i_am_a_file.txt"
+         * size: 0
+         * text: "data/bulk/wjriehl/subfolder/i_am_a_file.txt"
          */
-        function formatObjectDisplay(ret_obj) {
-            if (!ret_obj.id) {
-                return '';
+        function formatObjectDisplay(retObj) {
+            if (!retObj.id) {
+                return retObj.text || '';
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js:330]: This fixes the `Loading...` bit. Apparently every `Option` element in Select2 gets pushed through the formatter, including those that Select2 inserts itself. Seems a bit of a bug on their side, IMO, but this fixes it. Anything that gets used from a service to be formatted for display is expected to be in the form of an object with an `id` field anyway.

File: kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js [hunk 4]
```diff
@@ -332,53 +317,77 @@ define([
...
             } else if (dataSource === 'ftp_staging') {
-                if (!ret_obj.id) {
-                    return $('<div style="display:block; height:20px">').append(ret_obj.text);
+                if (!retObj.id) {
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js:332]: Removed inline css. Everything else in this function is as-is for now.

File: package.json [hunk 5]
```diff
@@ -157,7 +157,7 @@
         "requirejs-json": "0.0.3",
         "requirejs-plugins": "1.0.2",
         "select2": "4.0.13",
-        "select2-bootstrap-theme": "0.1.0-beta.9",
+        "select2-bootstrap-theme": "0.1.0-beta.10",
```
Comment (briehl) [inline package.json:160]: Bumped this to the latest version. Incidentally, we're not actually installing it right during runtime, so it's not even in use. But that's a separate issue.

**Review by ialarmedalien — APPROVED** [review]
👍

File: package.json (hunk 5 above)
Comment (ialarmedalien) [reply package.json:160]: 😖

**Pull request merged**